app/src/main/java/com/portfolio/tasky/usecases/NamePatternValidatorImpl.kt

```diff
@@ -0,0 +1,9 @@
+package com.portfolio.tasky.usecases
+
+import javax.inject.Inject
+
+class NamePatternValidatorImpl @Inject constructor() : NamePatternValidator {
+    override fun isValidName(name: CharSequence?): Boolean {
+        return Regex("^(?=.*[a-z]).{3,}\$").matches(name as CharSequence)
```

Comment:
In this case, since you are going to use your own regex, you don't need to have a NamePatternValidator and a NamePatternValidatorImpl
Since it's fully Kotlin code, nothing related to Android

Also, no need to use regex

You could simply check for stuff like "isNotEmpty" and "Length > 5" or similar stuff